pcstac/pcstac/main.py

```diff
@@ -75,6 +76,15 @@
 
 add_timeout(app, app_settings.request_timeout)
 
+
+@app.middleware("http")
```

Comment:
Unfortunately, I think the `middleware` decorator is just a wrapper around BaseHTTPMiddlware, so this doesn't effectively remove that class. Could we instead create our own Starlette middleware class that doesn't inherit from BaseHTTPMiddlware?

See https://kisspeter.github.io/fastapi-performance-optimization/middleware for a description of the difference and an [example implementation ](https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/api/stac_fastapi/api/middleware.py#L46)of that type of middleware from stacfastapi